llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -131,13 +131,14 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST) {
 
   getActionDefinitionsBuilder(G_BRCOND).legalFor({sXLen}).minScalar(0, sXLen);
 
+  getActionDefinitionsBuilder(G_BRJT).legalFor({{p0, sXLen}});
```

Comment:
I wanted to respond to your comment from https://github.com/llvm/llvm-project/pull/71951
 
> Does the s32 case ever occur? IRTranslator::emitJumpTableHeader seems to use the integer size of the pointer to create the type for that operand.

It would be fine to only support sXLen, although it definitely would be legal if we used s32 since it'll always generate a valid offset. I am okay to omit s32 as a legal type here to simplify. I don't have a strong opinion so you can decide.

Reply:
If it generated s32, the instruction selection code would have to insert an extend to use the index wouldn't it?

Reply:
Yea, sounds like it is better to not legalize it to avoid doing the extra work in selection to extend.